bug: shepherd does not comment on GitHub issue when rate-limit abort prevents builder startup

## Problem

When the shepherd detects a rate-limit abort **at builder startup** (before `run_phase_with_retry` returns), it exits immediately without posting a diagnostic comment on the GitHub issue. The `abandonment_info` path (added in #2891 / issue #2839) only fires when the builder phase fails *during* execution — not when it is rejected before starting.

## Observed Sequence

During the shepherd run for issue #2839:

```
[2026-02-18T22:43:22Z] [ERROR] API rate limit exceeded (threshold: 99%)
[2026-02-18T22:43:26Z] [INFO] Label cleanup: PR #848 is already merged — not reverting issue #2839 labels
[2026-02-18T22:43:26Z] [OK] Released claim for issue #2839
```

The shepherd hit the rate-limit check (in `BuilderPhase._check_rate_limit`) before invoking Claude and exited with code 1 (RATE_LIMIT_ABORT). No comment was posted on the issue.

This is distinct from the fix in #2891: that PR ensures comments are posted when `run_phase_with_retry` returns a non-retryable `PhaseResult`. The rate-limit pre-check happens earlier and returns the exit code directly.

## Expected Behavior

Any path that exits with a non-retryable error and reverts `loom:building` → `loom:issue` should post a diagnostic comment. Operators need to know why the issue was returned to the pool.

## Scope

Look for early-exit paths in the builder phase that bypass `run_phase_with_retry`:
- `_check_rate_limit()` in `BuilderPhase` (`phases/builder.py`)
- Any other pre-flight checks that return a non-zero exit code before the builder subprocess is launched

These should either:
1. Set `ctx.abandonment_info` before returning the exit code so `_post_fallback_failure_comment` picks it up, or
2. Post the GitHub comment directly from the pre-flight check

## References

- Observed during shepherd run for issue #2839 (task `163da3a`)
- Rate-limit check: `loom-tools/src/loom_tools/shepherd/phases/builder.py` (search `_check_rate_limit`)
- Comment posting: `loom-tools/src/loom_tools/shepherd/cli.py` (`_post_fallback_failure_comment`)
- Related fix: PR #2891 (issue #2839)

--- a/loom-tools/src/loom_tools/shepherd/cli.py
+++ b/loom-tools/src/loom_tools/shepherd/cli.py
@@ -2251,6 +2251,16 @@ def _post_fallback_failure_comment(ctx: ShepherdContext, exit_code: int) -> None
                 "Wait for the usage or plan limit to reset, or re-authenticate "
                 "with a different plan before retrying."
             )
+        elif failure_data.get("api_rate_limited"):
+            failure_mode = (
+                "API rate limit pre-check — usage exceeded configured threshold "
+                "before builder was started"
+            )
+            safe_to_retry = True
+            advice = (
+                "The issue has been returned to `loom:issue`. "
+                "Retry after the rate limit resets (usually a few minutes)."
+            )
         else:
             # Generic builder failure (MCP failure, low-output, etc.)
             failure_mode = message or "unexpected failure"

--- a/loom-tools/src/loom_tools/shepherd/phases/builder.py
+++ b/loom-tools/src/loom_tools/shepherd/phases/builder.py
@@ -363,6 +363,7 @@ class BuilderPhase:
                 status=PhaseStatus.FAILED,
                 message=f"API rate limit exceeded (threshold: {ctx.config.rate_limit_threshold}%)",
                 phase_name="builder",
+                data={"api_rate_limited": True},
             )
 
         # Report phase entry

--- a/loom-tools/tests/shepherd/test_cli.py
+++ b/loom-tools/tests/shepherd/test_cli.py
@@ -4393,6 +4393,38 @@ class TestPostFallbackFailureComment:
         assert "usage/plan limit" in body
 
     @patch("subprocess.run")
+    def test_abandonment_info_api_rate_limited(self, mock_run: MagicMock) -> None:
+        """When abandonment_info has api_rate_limited, posts pre-check rate-limit comment.
+
+        Covers the path where _is_rate_limited() fires before builder startup
+        (issue #2906).  The PhaseResult now carries data={"api_rate_limited": True}
+        which propagates into abandonment_info["failure_data"].
+        """
+        ctx = MagicMock()
+        ctx.config.issue = 42
+        ctx.config.task_id = "pqr1234"
+        ctx.repo_root = Path("/fake/repo")
+        ctx.abandonment_info = {
+            "phase": "builder",
+            "exit_code": 1,
+            "failure_data": {
+                "api_rate_limited": True,
+            },
+            "message": "API rate limit exceeded (threshold: 99%)",
+            "task_id": "pqr1234",
+        }
+
+        _post_fallback_failure_comment(ctx, ShepherdExitCode.BUILDER_FAILED)
+
+        mock_run.assert_called_once()
+        cmd = mock_run.call_args[0][0]
+        body_idx = cmd.index("--body") + 1
+        body = cmd[body_idx]
+        assert "Shepherd abandoned issue" in body
+        assert "API rate limit pre-check" in body
+        assert "Retry after the rate limit resets" in body
+
+    @patch("subprocess.run")
     def test_abandonment_info_generic_failure(self, mock_run: MagicMock) -> None:
         """When abandonment_info has no specific flag, uses message as failure mode."""
         ctx = MagicMock()

--- a/loom-tools/tests/shepherd/test_phases.py
+++ b/loom-tools/tests/shepherd/test_phases.py
@@ -1328,6 +1328,34 @@ class TestBuilderPhase:
         assert mock_context.pr_number == 302
         mock_cleanup.assert_not_called()
 
+    def test_rate_limit_precheck_sets_api_rate_limited_flag(
+        self, mock_context: MagicMock
+    ) -> None:
+        """Pre-flight rate-limit check returns data with api_rate_limited=True.
+
+        When _is_rate_limited() fires before builder startup, the returned
+        PhaseResult must carry data={"api_rate_limited": True} so that the
+        orchestrator's abandonment_info records a recognisable flag and
+        _post_fallback_failure_comment generates a specific comment instead of
+        the generic "unexpected failure" fallback.  See issue #2906.
+        """
+        mock_context.check_shutdown.return_value = False
+        mock_context.config.rate_limit_threshold = 99
+
+        builder = BuilderPhase()
+
+        with (
+            patch(
+                "loom_tools.shepherd.phases.builder.get_pr_for_issue", return_value=None
+            ),
+            patch.object(builder, "_is_rate_limited", return_value=True),
+        ):
+            result = builder.run(mock_context)
+
+        assert result.status == PhaseStatus.FAILED
+        assert result.data.get("api_rate_limited") is True
+        assert "rate limit" in result.message.lower()
+
 
 class TestBuilderCountCommitsAheadOfMain:
     """Unit tests for BuilderPhase._count_commits_ahead_of_main."""

